src/uu/ls/src/ls.rs

```diff
@@ -3019,7 +3019,23 @@ fn display_file_name(
         let absolute_path = fs::canonicalize(&path.p_buf).unwrap_or_default();
         let absolute_path = absolute_path.to_string_lossy();
 
-        // TODO encode path
+        #[cfg(not(target_os = "windows"))]
+        let unencoded_chars = "_-.:~/";
+        #[cfg(target_os = "windows")]
+        let unencoded_chars = "_-.:~\\";
```

Comment:
I should've asked this on the previous review, but did you check GNU for the backslash? I think maybe Windows either needs both back and forward slash (because both are allowed) or only forward slash because that matches the spec (as explained on the wikipedia page: https://en.wikipedia.org/wiki/Percent-encoding). Although of course maybe the way GNU does it is different from the spec that wikipedia describes.

Reply:
GNU encodes backslashes. And as Windows uses backslashes in paths I thought it doesn't make sense to encode them on Windows. It's guess work from my side, as I don't know Windows. In the latest push I added the forward slash to `unencoded_chars`.